engine/access/rest/rest_api.go

```diff
@@ -0,0 +1,94 @@
+package rest
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// RestAPIHandler provides the implementation of each of the REST API
+type RestAPIHandler struct {
+	backend *backend.Backend
+	logger  zerolog.Logger
+	encoder encoding.Encoder
+}
+
+func NewRestAPIHandler(backend *backend.Backend, logger zerolog.Logger) *RestAPIHandler {
+	return &RestAPIHandler{
+		backend: backend,
+		logger:  logger,
+		encoder: encoding.DefaultEncoder, //use the default JSON encoder
+	}
+}
+
+func (restAPI *RestAPIHandler) BlocksIdGet(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	vars := mux.Vars(r)
+	idParam, ok := vars["id"]
+	if !ok {
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
we should tell the user why the request was invalid

Reply:
Yep, we can use `errorResponse` here.

Reply:
In fact, I *think* this should actually never happen, right? Because if the routes are working properly, we should not hit this handler unless the `id` part of `/blocks/{id}` is nonempty?

Can we confirm this, and if so, we probably don't even need to capture the `ok` since we're guaranteed it will always be true?

Reply:
confirmed - a request without an ID does not reach here. Changing the code to assume vars["id"] will always returning a value.